src/collection.js

```diff
@@ -626,13 +629,27 @@ export default class Collection {
     options = Object.assign({
       strategy: Collection.strategy.MANUAL,
       lastModified: this.lastModified,
+      limit: options.fetchLimit || null,
+      token: options.fetchToken || null,
       headers: {},
     }, options);
     // First fetch remote changes from the server
     return this.api.fetchChangesSince(this.bucket, this.name, {
       lastModified: options.lastModified,
+      limit: options.limit,
+      token: options.token,
       headers: options.headers
     })
+      // Deal with pagination
+      .then(changes => {
+        if (changes.fetchToken) {
+          syncResultObject.fetchToken = changes.fetchToken;
+        }
+        // TODO: deal with the case where records disappear from
+        // pages already fetched, due to updates to the server
+        // while the paginating query series is in progress.
```

Comment:
As far as I can remember we should get a 412 from the server whenever this happens. @Natim can you confirm this?